### DISCUSSION TITLE
2FA page improvements: Rename Google Authenticator and drop Authy

### DISCUSSION THREAD

**Discussion by iliaross** [webmin/webmin#2260] [Ideas]
Hey Jamie!

1. I think we don't need to mention Google Authenticator when referring to OTP providers.

2. Also, Authy has changed significantly since the time when support for it was initially added in Webmin. Recently, Twilio, the company behind Authy, dropped support for all desktop applications earlier this year and now only offer mobile apps. I tried registering an account with Authy, and it seems to work similarly to OTP, requiring a QR code scan. Though, to make it work, we need to have [twilio-cli](https://github.com/twilio/twilio-cli), but it doesn’t seem to be available in the repositories.

**Comment by jcameron** [comment]
Hey Ilia - are there other TOPT apps similar to google authenticator that implement the same protocol?

Sounds like we should drop or at least hide Authy then. I wonder if there are other options we could offer instead, like security keys?

**Reply by iliaross** [reply]
> Are there other TOPT apps similar to google authenticator that implement the same protocol?

Yeah, that's right! Plenty of them.

> Sounds like we should drop or at least hide Authy then.

Agreed! I think we should just hide it and leave only one option to enable TOPT.

> I wonder if there are other options we could offer instead, like security keys?

I don't know .. How about recovery phrase?

**Reply by jcameron** [reply]
Ok I'll look into this.

I'm thinking of other potential options for second-factor logins, not just account recovery. SMS would be nice, but I can't see how that could be done for free..

**Reply by iliaross** [reply]
> SMS would be nice, but I can't see how that could be done for free.

SMS isn't ideal for security because we don’t have full control over our cell phone numbers, making it vulnerable to man-in-the-middle attacks and interception.

A better approach is to use 2FA with a special recovery code that appears if a one-time password is entered incorrectly a few times.

**Reply by jcameron** [reply]
Actually ideal would be something like a security key or passkey, but I'm not sure how to integrate with those..

**Reply by iliaross** [reply]
> Actually ideal would be something like a security key or passkey, but I'm not sure how to integrate with those..

Modern browsers support WebAuthn. We could use its API ..?

**Reply by jcameron** [reply]
I've renamed Google Authenticator to just TOTP in Webmin here : https://github.com/webmin/webmin/commit/d1ec3910d226e72b864577e82b0393614873818d

Another provider we could consider supporting is Duo.

**Reply by iliaross** [reply]
> I've renamed Google Authenticator to just TOTP in Webmin here : [d1ec391](https://github.com/webmin/webmin/commit/d1ec3910d226e72b864577e82b0393614873818d)

Got it, thanks! I've update translations!

> Another provider we could consider supporting is Duo.

I think the free version of Duo isn’t much different from the existing TOTP, is it?

**Reply by jcameron** [reply]
Yeah it does look like Authy isn't accepting new users, but I will keep it around in Webmin since existing users are still supported.

**Reply by iliaross** [reply]
Perhaps, we shouldn't display it for new setups?
